internal/ignition/ignition.go

```diff
@@ -276,12 +307,14 @@ func NewBuilder(log logrus.FieldLogger, staticNetworkConfig staticnetworkconfig.
 }
 
 // NewGenerator returns a generator that can generate ignition files
-func NewGenerator(workDir string, installerDir string, cluster *common.Cluster, releaseImage string, releaseImageMirror string,
-	serviceCACert, installInvoker string, s3Client s3wrapper.API, log logrus.FieldLogger, operatorsApi operators.API,
+func NewGenerator(db *gorm.DB, serviceBaseURL string, workDir string, installerDir string, cluster *common.Cluster, releaseImage string, releaseImageMirror string,
```

Comment:
with so many parameters maybe we should consider having a configuration struct?

Reply:
Indeed a lot of parameters, however, all this function is doing is constructing those parameters to a `generator` struct and returning that struct.

Reply:
Therefore I'm not sure we want to create an interim struct just so we can feed that function and return a new generator struct, which will be identical.
Let me know if I missed something here.

Reply:
the difference between a function parameters and a struct is that it's clear to see what is set in a struct. 
it's a common pattern in apis that can have a lot of parameters. 
It's out of context for this PR so you can resolve this comment